spec/Types.tex

```diff
@@ -94,6 +95,20 @@ \subsection{The Void Type}
 There may be storage associated with a value of type \chpl{void}, in which
 case its lifetime obeys the same rules as a value of type \chpl{int}.
 
+\subsection{The Nothing Type}
+\label{The_Nothing_type}
+\index{nothing@\chpl{nothing}}
+\index{types!nothing@\chpl{nothing}}
+
+The \chpl{nothing} type is used to indicate a variable or field that
+should be removed by the compiler. The value \chpl{none} is the only
+value of type \chpl{nothing}.
```

Comment:
It seems to me that somewhere in the spec it should say something about the different rules about returning `none` and `nothing` or assigning one to a variable - since these are different for `nothing` and `void`.

Did you intentionally not put `none` into this spec change?

Reply:
I'll look into adding something for rules about assigning/returning none.

Reply:
I added a description of what assigning `none` to a variable does.

Reply:
Hmm, it still seems that we need to say somehow that
``` chapel
proc f() { }
var x = f();
```

is an error (can't store void value into a variable).

Reply:
I added a sentence stating that you can't store a void function's result into a variable.